ppr-ui/src/components/mhrRegistration/ReviewConfirm/SubmittingPartyReview.vue

```diff
@@ -16,34 +16,137 @@
+                class="content"
+                :schema="addressSchema"
+                :value="address"
+              >
+              </base-address>
+              <p v-else class="content"> (Not Entered) </p>
+            </v-col>
+            <v-col cols="3" class="pl-3">
+              <p class="content">{{getMhrRegistrationSubmittingParty.emailAddress || '(Not Entered)'}}</p>
+            </v-col>
+            <v-col cols="3" class="pl-4">
+              <p class="content" v-html="parsePhoneNumber()"></p>
+            </v-col>
+          </v-row>
+
+          <div class="px-4">
+            <v-divider />
+          </div>
+
+          <v-row no-gutters class="px-6 py-7">
+            <v-col cols="3">
+              <p class="side-header">Attention or<br>Reference Number</p>
+            </v-col>
+            <v-col cols="9">
+              <p class="content">{{getMhrAttentionReferenceNum || '(Not Entered)'}}</p>
+            </v-col>
+          </v-row>
         </section>
       </template>
     </div>
   </v-card>
 </template>
 
 <script lang="ts">
-import { defineComponent, reactive, toRefs } from '@vue/composition-api'
+import { computed, defineComponent, reactive, toRefs } from '@vue/composition-api'
 import { RouteNames } from '@/enums'
+import { useGetters } from 'vuex-composition-helpers'
+import { BaseAddress } from '@/composables/address'
+import { PartyAddressSchema } from '@/schemas'
 
 export default defineComponent({
   name: 'SubmittingPartyReview',
-  components: {},
+  components: {
+    BaseAddress
+  },
   props: {},
   setup () {
-    const localState = reactive({})
+    const {
+      getMhrRegistrationSubmittingParty,
+      getMhrAttentionReferenceNum
+    } = useGetters<any>([
+      'getMhrRegistrationSubmittingParty',
+      'getMhrAttentionReferenceNum'
+    ])
+
+    const localState = reactive({
+      address: computed(() => getMhrRegistrationSubmittingParty.value.address),
+      businessName: computed(() => getMhrRegistrationSubmittingParty.value.businessName),
+      personName: computed(() => getMhrRegistrationSubmittingParty.value.personName),
+      hasAddress: computed(() => !Object.values(localState.address).every(val => !val))
+    })
+
+    const addressSchema = PartyAddressSchema
+
+    const parsePhoneNumber = () => {
+      const phone = getMhrRegistrationSubmittingParty.value
+      const phoneNum = phone.phoneNumber
+      const ext = phone.phoneExtension ? ' &nbsp;Ext ' + phone.phoneExtension : ''
+      return phoneNum ? `${phoneNum}${ext}` : '(Not Entered)'
+    }
+
+    const getSubmittingPartyName = () => {
+      if (!localState.businessName && Object.values(localState.personName).every(val => !val)) {
+        return '(Not Entered)'
+      } else {
+        return localState.businessName
+          ? localState.businessName
+          : localState.personName.middle
+            ? localState.personName.first + ' ' + localState.personName.middle + ' ' + localState.personName.last
```

Comment:
As per best practices, we try to avoid having nested ternary operators. Also, I realized that the middle name will always be defined, as in the `state-model`, `middle: ''`.

Reply:
replaced the first ternary with an if statement!